### ISSUE TITLE
[MPS] TensorIterator should take advantage of contiguous inner dimension even if tensor is non-contiguous as a whole

### ISSUE THREAD

**Issue by jhavukainen** [pytorch/pytorch#188482]
### 🐛 Describe the bug

Follow-up to discussion at: https://github.com/pytorch/pytorch/pull/188200. 

Elementwise unary/binary ops and same-dtype copies on a tensor that is contiguous in its innermost dimension but non-contiguous overall fall back to the fully generic per-element strided kernel. For example

```python
 y = torch.exp(x[:, :W]) # non-contiguous argument -> hits the per-element kernel
```

Yet the innermost run is a clean unit stride span that could move as wide coalesced vectors. This leaves a lot of bandwidth on the table.

Opening a PR with a suggestion: a new kernel flavor that detects this case via TensorIterator's existing contiguity info and dispatches a vectorized path benefiting the elementwise metal kernels built on the iterator.

### Versions

nightly, M2 Max, macOS27 beta

cc @kulinseth @malfet @DenisVieriu97 @aditvenk @Isalia20
